src/Avalonia.Base/Input/KeyGesture.cs

```diff
@@ -95,8 +96,28 @@ public static KeyGesture Parse(string gesture)
             return new KeyGesture(key, keyModifiers);
         }
 
-        public override string ToString()
+        public override string ToString() => ToString(null, null);
+
+        /// <summary>
+        /// Returns the current KeyGesture as a string formatted according to the format string and appropriate IFormatProvider
+        /// </summary>
+        /// <param name="format">The format to use. 
+        /// <list type="bullet">
+        /// <item><term>null or "" or "g"</term><description>The Invariant format, uses Enum.ToString() to format Keys.</description></item>
+        /// <item><term>"p"</term><description>Use platform specific formatting as registerd.</description></item>
+        /// </list></param>
+        /// <param name="formatProvider">The IFormatProvider to use.  If null, uses the appropriate provider registered in the Avalonia Locator, or Invariant.</param>
+        /// <returns>The formatted string.</returns>
+        /// <exception cref="FormatException">Thrown if the format string is not null, "", "g", or "p"</exception>
+        public string ToString(string? format, IFormatProvider? formatProvider)
         {
+            var formatInfo = format switch
+            {
+                null or "" or "g" => KeyGestureFormatInfo.Invariant,
+                "p" => KeyGestureFormatInfo.GetInstance(formatProvider),
+                _ => throw new FormatException("Unknown format specifier")
+            };
```

Comment:
We probably should consider using CultureInfo passed as IFormatProvider to be used for localization here. And pass it down from IValueConverter. But that's a bigger issue for another day.

Reply:
True, I imagine "Up Arrow" is not going to be universally understood by every language on Earth.  I'll take a look into how that sort of thing is done, but as you say it's a bigger problem for another day.  I'll just take steps to try and make sure I'm not making that other day any harder than it needs to be.